resharper/resharper-unity/src/Unity/CSharp/Daemon/UsageChecking/UsageInspectionsSuppressor.cs

```diff
@@ -59,7 +59,10 @@ public bool SuppressUsageInspectionsOnElement(IDeclaredElement element, out Impl
                                      IsUxmlFactory(cls):
                     flags = ImplicitUseKindFlags.InstantiatedNoFixedConstructorSignature;
                     return true;
-
+                case IStruct @struct when unityApi.IsUnityType(@struct) ||
```

Comment:
1. Why only struct? System also could be class
2. Why are you setting implicit use for any "IsUnityType"?

Reply:
Class is processed on the line 55